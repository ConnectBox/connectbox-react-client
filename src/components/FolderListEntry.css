.subfolder-icon-container {
    margin-left: 5px;
    width: 14px;
    height: 14px;
    display: inline-block;
    white-space: nowrap;
}

.subfolder-icon-container img {
  display: block;
  width: 100%;
  height: auto;
}
