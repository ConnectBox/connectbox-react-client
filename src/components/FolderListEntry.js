import './FolderListEntry.css'

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {  topLevelCategory, mediaTypes } from '../utils'

class FolderListEntry extends Component {
    static propTypes = {
        file: PropTypes.object.isRequired,
        folderPath: PropTypes.string.isRequired,
        contentRoot: PropTypes.string.isRequired,
        iconMetadata: PropTypes.object.isRequired
    }

  render () {
    const { file, contentRoot, folderPath, iconMetadata } = this.props
    const { icon_file, name } = file
    const fileLink = `${file.type === 'directory' ? '#' : contentRoot}${folderPath}/${name}`

    let spanClass = file.type === 'directory' ? 'directory' : ''

    if (icon_file) {
      return (
          <tr>
            <td>
              <a href={fileLink}>
                <div className="subfolder-icon-container">
                    <img alt="" src={`${contentRoot}${folderPath}/${icon_file}`} />
                </div>
                <span className={spanClass}>
                    <span>&nbsp;{name}</span>
                </span>
              </a>
            </td>
          </tr>
      )
    }

    return (
      <tr>
        <td>
          <a href={fileLink}>
            <i className={ `fa ${topLevelCategory(file, iconMetadata)}` }></i>
            <i className={`fa ${mediaTypes(file)}`}></i>
            <span className={spanClass}>
                <span>&nbsp;{name}</span>
            </span>
          </a>
        </td>
      </tr>
    )
  }
}

export default FolderListEntry
