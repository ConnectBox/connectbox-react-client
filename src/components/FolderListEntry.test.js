/* global it, expect, */
import React from 'react'
import FolderListEntry from './FolderListEntry'
import renderer from 'react-test-renderer'

it('Render FolderListEntry', () => {
  const iconMetadata = {ids: [], names: {}}
  const file = {type: 'file', ext: 'doc', name: 'Test File'}
  const component = renderer.create(
    <FolderListEntry
      contentRoot='/content'
      key='fooEntry0'
      file={file}
      folderPath='foo'
      iconMetadata={iconMetadata} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
