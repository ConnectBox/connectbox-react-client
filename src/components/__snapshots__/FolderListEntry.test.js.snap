// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FolderListEntry 1`] = `
<tr>
  <td>
    <a
      href="/contentfoo/Test File"
    >
      <i
        className="fa "
      />
      <i
        className="fa fa-file-text-o"
      />
      <span
        className=""
      >
        <span>
          Â 
          Test File
        </span>
      </span>
    </a>
  </td>
</tr>
`;
